src/htmldocvisitor.cpp

```diff
@@ -401,7 +401,8 @@ void HtmlDocVisitor::operator()(const DocURL &u)
   else // web address
   {
     m_t << "<a href=\"";
-    m_t << u.url() << "\">";
+    filter(u.url());
```

Comment:
The url needs URL encoding inside the href and HTML encoding outside, see https://stackoverflow.com/a/1812486/784672. Probably also holds for XML and docbook.

Reply:
Looks indeed the case that is required for all formats that potentially use the string to start a HTML browser, so probably not only XML and Docbook but also LaTeX, RTF, ...

Reply:
Hmm, I tried URL encoding, but then & is not escaped, so it still doesn't pass XML/XHTML checks. So I think your fix is correct as it is.